rust/processor/src/gap_detectors/mod.rs

```diff
@@ -28,18 +28,26 @@ pub enum GapDetector {
 
 #[enum_dispatch(GapDetectorTrait)]
 pub enum GapDetectorResult {
-    DefaultGapDetectorResult,
-    ParquetFileGapDetectorResult,
+    DefaultGapDetectorResult(DefaultGapDetectorResult),
+    ParquetFileGapDetectorResult(ParquetFileGapDetectorResult),
 }
+
+pub trait GapDetectorTrait: Send {
+    fn process_versions(
```

Comment:
For traits we should have comments on the trait and the methods of that trait explaining how implementations of the trait should behave.